src/response.ts

```diff
@@ -575,29 +575,54 @@ export class Response implements DenoResponse {
     return this.send(body);
   }
 
-  // TODO: back-compat support for Express signature.
-  // Namely objects and arrays.
   /**
    * Set header `field` to `value`, or pass
    * an object of header fields.
    *
    * Examples:
    *
-   *    res.set('Accept', 'application/json');
-   *
+   *     res.set('Accept', 'application/json');
+   *     res.set({
+   *       'Accept-Language': 'en-US,en;q=0.5',
+   *       'Accept': 'text/html',
+   *     });
    * @param {string} field
    * @param {string} value
    * @return {Response} for chaining
    * @public
    */
-  set(field: string, value: string): this {
-    const lowerCaseField = field.toLowerCase();
+  set(field: string, value: string): this;
+  set(obj: Record<string, string[] | string>): this;
+  set(field: string, value: string[]): this;
+  set(field: unknown, value?: unknown): this {
+    if (arguments.length === 2) {
+      const lowerCaseField = (field + "").toLowerCase();
+      const firstVal = (Array.isArray(value) ? value[0] : value) + "";
```

Comment:
Can we use type casting to assign this values as `string` instead of using `+ ""`?